### ISSUE TITLE
ProhibitLeadingZeroes exception for bitwise operators

### ISSUE THREAD

**Issue by epa** [Perl-Critic/Perl-Critic#577]
ValuesAndExpressions::ProhibitLeadingZeros has various exceptions where giving an octal value as 0644 or similar is the normal idiom.  They are summarized at the top of its documentation.  One more case that should be allowed is octal constants with bitwise operators:

```
use File::stat;
my $st = stat 'file';
my $mode = $st->mode & 0777;
```

When an octal constant is on either side of the bitwise | or & operators the policy should not complain.

**Comment by Julien-Elie** [comment]
I agree it would be useful to allow leading zeroes in bitwise operations.  Maybe with an allow option?
